src/app/store/controller/selectors.ts

```diff
@@ -373,6 +374,34 @@ const getByFabricId = createSelector(
       }, [])
 );
 
+/**
+ * Get all region controllers and separate by their vault configuration status.
+ * @param state - The redux state.
+ * @returns Two lists of region controllers - one where none are configured with vault, the other where all are configured with vault.
+ */
+const getVaultConfiguredControllers = createSelector(
+  [defaultSelectors.all],
+  (controllers: Controller[]) => {
+    const regionControllers = controllers.filter((controller) => {
+      return (
+        controller.node_type === NodeType.REGION_CONTROLLER ||
+        controller.node_type === NodeType.REGION_AND_RACK_CONTROLLER
+      );
+    });
+    const unconfiguredControllers = regionControllers.filter((controller) => {
+      return (
+        controller.vault_configured === false ||
+        controller.vault_configured === undefined
```

Comment:
Most of the time it should just return false if Vault hasn't been configured, but if it returns `undefined` that also means that this controller is not configured.